### ISSUE TITLE
Fix false positives for spaceless multiplication and division in calc functions in unit-no-unknown

### ISSUE THREAD

**Issue by alexander-schranz** [stylelint/stylelint#2551]
A calc throw a unit no unknown error.

> Which rule, if any, is this issue related to?

`unit-no-unknown`

> What CSS is needed to reproduce this issue?

```css
a { top: calc(1px+2px); }
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
      "unit-no-unknown": true
  }
}
```

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

```
npm run lint:css

   "lint:css": "stylelint Resources/css/*.scss"
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No error

> What actually happened (e.g. what warnings or errors you are getting)?

`Unexpected unknown unit "px+2px"`

**Comment by davidtheclark** [comment]
> it is breaking changes

I don't think so ... it's a bug fix. That's clearly not how the rule is meant to behave.

**Comment by evilebottnawi** [comment]
@davidtheclark some people can use `calc` with unknown unit (example using `postcss` plugin) https://github.com/Semigradsky/postcss-russian-units :smile:  or without units https://github.com/antyakushev/postcss-default-unit and it is break build

**Comment by hudochenkov** [comment]
Just for information:

`a { top: calc(1px+2px); }` is not a valid CSS.

> The + and - operators must always be surrounded by whitespace.
> The * and / operators do not require whitespace, but adding it for consistency is allowed, and recommended.

https://developer.mozilla.org/en-US/docs/Web/CSS/calc

I've checked:

```css
/* Not valid CSS */
a { top: calc(10px+2px); } /* Unexpected unknown unit "px+2px" */
a { top: calc(10px-2px); } /* Unexpected unknown unit "px-2px" */

/* Valid CSS */
a { top: calc(10px*2); } /* Unexpected unknown unit "px*2" */
a { top: calc(10px/2); } /* No errors */
```

**Comment by davidtheclark** [comment]
Ohhh, maybe this is not a bug then @hudochenkov, huh?

**Comment by hudochenkov** [comment]
@davidtheclark It's a bug for `a { top: calc(10px*2); }`. Valid CSS.

**Comment by alexander-schranz** [comment]
just mention it that the example `a { top: calc(1px+2px); }` is from the [function-calc-no-unspaced-operator/](https://github.com/stylelint/stylelint/blob/master/lib/rules/function-calc-no-unspaced-operator/README.md) rule.

**Comment by evilebottnawi** [comment]
@alexander-schranz trying to check invalid `css` is bad practice, it is will bring a big overload in the rules and instability, just validate your `css` (in `github` there are a lot of tools for this :+1: )

**Comment by alexander-schranz** [comment]
@evilebottnawi i use stylefmt in front of stylelint so I have no problems. And I also use the `function-calc-no-unspaced-operator` so this issue doesnt effect my process. I just report it because I think the rule dont behave like it should.

**Comment by evilebottnawi** [comment]
Issue related to `postcss-value-parser` https://github.com/TrySound/postcss-value-parser/issues/33

**Comment by jeddy3** [comment]
I've removed `status:blocked` label as per https://github.com/TrySound/postcss-value-parser/issues/33. This is a false positive bug, for `a { top: calc(10px*2); }` that we should deal with at our end.

This issue has `help wanted` and `beginner friendly` labels, so if there are any new contributors looking to help... please get stuck in :)

**Comment by CAYdenberg** [comment]
This can be closed as it was fixed in #2848 and the other cases are covered by #2875
